cpp/src/arrow/c/bridge.cc

```diff
@@ -739,19 +738,20 @@ Status ExportDeviceArray(const Array& array, RawSyncEvent sync_event,
   exporter.Finish(&out->array);
 
   auto* pdata = reinterpret_cast<ExportedArrayPrivateData*>(out->array.private_data);
-  pdata->sync_event_ = sync_event;
-  out->sync_event = sync_event.sync_event;
+  pdata->sync_ = sync;
```

Comment:
... then move into `pdata`
```suggestion
  pdata->sync_ = std::move(sync);
```